### DISCUSSION TITLE
Login Page Security

### DISCUSSION THREAD

**Discussion by countrymusicfy** [AzuraCast/AzuraCast#4228] [Ideas]
Regardless of whether one sets up a sub-domain for azuracast, when one posts a player on twitter, that subdomain displays to the public.  Which means that anyone can easily find the azuracast login page.

I realize that adding login security features is time-consuming, but how about just adding the option to re-name the login page? And having the default azuracast page be the public player page rather than the login page?

**Comment by ItsMitchh** [comment]
I don't believe this raises a legitimate security concern, as there's no way to my knowledge to brute force accounts, we use Argon as a password hashing mechanism so it'll be near impossible to breach an account, unless your account password is incredibly basic and or easy to guess. 

We do advise people use common knowledge when creating passwords and don't re-use them across multiple sites.

**Comment by BusterNeece** [comment]
We actually _do_ already offer a mitigation to this issue: you can, via the system administration settings, provide a custom URL where the main site root (the `yoursite.com/` with no path) redirects to. That means any users of the system will have to explicitly visit the `/login` page (or `/dashboard`, or some other page that requires authentication) to get into AzuraCast, but it prevents most people from ever seeing the login screen on casual exploration.

As for the security of the login process, we do indeed use today's best practices in hashing and storing passwords, specifically the Argon2ID algorithm which has each password individually salted and is intentionally computationally-intensive to resist brute-forcing passwords, even if the hashed versions are leaked for any reason. We also implement a TOTP-based two-factor authentication that we strongly recommend any system-wide administrators use if they can. Together, the security of both are excellent.
